src/main/java/com/nike/cerberus/operation/core/RollingRebootWithHealthCheckOperation.java

```diff
@@ -0,0 +1,225 @@
+                                              final int numConsecutiveResponsesExpected) {
+
+        int responseCode;
+        int consecutiveResponses = 0;
+        while (consecutiveResponses < numConsecutiveResponsesExpected) {
+
+            responseCode = executeHealthCheck(healthCheckUrl);
+
+            if (responseCode == expectedStatusCode) {
+                consecutiveResponses++;
+            } else if (consecutiveResponses > 0) {
+                final String message = Chalk.on("Instance health check did not repeat response code ({}), {} times").red().bold().toString();
+                logger.debug(message, expectedStatusCode, numConsecutiveResponsesExpected);
+                consecutiveResponses = 0;
+            }
+
+            try {
+                TimeUnit.SECONDS.sleep(NUM_SECS_BETWEEN_HEALTH_CHECKS);
+            } catch (InterruptedException ie) {
+                logger.error(Chalk.on("Timeout between health checks has been interrupted").red().bold().toString());
+                return;
+            }
+        }
+    }
+
+    /**
+     * Execute the given health check
+     * @param healthCheckUrl - Name of that EC2 instance belongs to
+     * @return - Response code of the health check
+     */
+    private int executeHealthCheck(final String healthCheckUrl) {
+
+        final OkHttpClient okHttpClient = new OkHttpClient.Builder()
+                .hostnameVerifier(new NoopHostnameVerifier())
+                .proxy(proxy)
+                .connectTimeout(DEFAULT_HTTP_TIMEOUT, DEFAULT_HTTP_TIMEOUT_UNIT)
+                .writeTimeout(DEFAULT_HTTP_TIMEOUT, DEFAULT_HTTP_TIMEOUT_UNIT)
+                .readTimeout(DEFAULT_HTTP_TIMEOUT, DEFAULT_HTTP_TIMEOUT_UNIT)
+                .build();
+
+        Request requestBuilder = new Request.Builder()
+                .url(healthCheckUrl)
+                .get()
+                .build();
+
+        try {
+            final Response response = okHttpClient.newCall(requestBuilder).execute();
+            logger.debug("Health check returned status: {}, URL: {}", response.code(), healthCheckUrl);
+            int responseCode = response.code();
+            response.close();
+            return responseCode;
+        } catch (IOException ioe) {
+            final String message = Chalk.on("Health check failed, Cause: {}, URL: {}").red().toString();
+            logger.debug(message, ioe.getMessage(), healthCheckUrl);
+        }
+
+        return HEALTH_CHECK_FAILED_CODE;
+    }
+
+    @Override
+    public boolean isRunnable(final RollingRebootWithHealthCheckCommand command) {
+
+        final StackName stackName = command.getStackName();
+        final String stackId = configStore.getStackId(stackName);
+        final Map<String, String> stackParameters = cloudFormationService.getStackParameters(stackId);
+
+        if (HEALTH_CHECK_MAP.containsKey(stackName.getName()) && stackParameters.containsKey(MIN_INSTANCES_STACK_PARAMETER_KEY)) {
+            return true;
+        }
+
```

Comment:
nitpick: I'd put this last clause in an else statement.  It just makes it slightly more obvious and more like http://wiki.c2.com/?SingleFunctionExitPoint